filebeat/docs/load-balancing.asciidoc

```diff
@@ -21,6 +21,13 @@ output.logstash:
 The `loadbalance` option is available for Redis, Logstash, and Elasticsearch
 outputs. The Kafka output handles load balancing internally.
 
+Beat’s loadbalance: true causes it to send data if-and-only-if it can send to all hosts in the list.
+
+When loadbalance: false, Filebeat will pick a host at random from the list and send data, failing over to other entries. It also accepts a ttl, which will cause it to voluntarily reestablish a connection to a new host after a time even if it is healthy.
+As long as the first chosen endpoint is available, then it continues to communicate with that one. It only switches to another endpoint in case of failure.
+
+When loadbalance: true is set, Filebeat will connect to all hosts in the list and evenly distribute the events to them. This has the side effect of blocking publishing to all hosts if any one of them is down or exerting back-pressure.
+
```

Comment:
Thanks! I'll find out who else can review.

One (probably dumb) question. To me these two seem contradictory:

`Filebeat also accepts a time to live (TTL), which causes it to voluntarily
reestablish a connection to a new host after a time even if it is healthy.`

`As long as the first chosen endpoint is available, Filebeat continues to
communicate with it. Filebeat only switches to another endpoint in case of a
failure.`

Is the second case maybe if a TTL is not set? That is, if no TTL is set Filebeat continues to communicate with the healthy endpoint?

Reply:
Yes indeed it is a bit contradictory, if no TTL is set Filebeat continues to communicate with the healthy endpoint.
Otherwise it does not change until the current endpoint fails

Reply:
I'll commit the change and find another reviewer.